tests/unit/db/account_tests.py

```diff
@@ -0,0 +1,640 @@
+#!/usr/bin/env python
+# Copyright (c) 2015 IBM. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+_account_tests_
+
+account module - Unit tests for CouchDB and Cloudant account classes
+
+Only CouchDB tests are set to execute whereas the Cloudant tests are
+turned off by default.
+
+In order to run Cloudant tests the following environment variables
+must be set:
+
+- RUN_CLOUDANT_TESTS: Set this to anything to trigger Cloudant
+  tests to run.
+  example: export RUN_CLOUDANT_TESTS=1
+- CLOUDANT_ACCOUNT: Set this to the Cloudant account that you wish
+  to connect to.
+  example: export CLOUDANT_ACCOUNT=account
+- CLOUDANT_USER: Set this to the username to connect to the account
+  with.
+  example: export CLOUDANT_USER=user
+- CLOUDANT_PASSWORD: Set this to the password for the username
+  specified.
+  example: export CLOUDANT_PASSWORD=password
+- CLOUDANT_URL: Optionally set this to override the construction of
+  Cloudant URL.
+  example: export CLOUDANT_URL=http://account.cloudant.com
+
+"""
+
+import unittest
+import requests
+import json
+import base64
+import os
+import uuid
+from datetime import datetime
+
+from cloudant.account import CouchDB, Cloudant
+from cloudant.errors import CloudantException
+
+class CouchDBAccountTests(unittest.TestCase):
+    """
+    CouchDB Class unit tests
+    """
+
+    @classmethod
+    def setUpClass(self):
+        """
+        Set up CouchDB instance to be used by all tests
+        """
+        admin_ep = '{0}/_config/admins'.format(
+            os.environ.get('COUCHDB_HOST', '127.0.0.1:5984')
+            )
+        user = os.environ.get('COUCHDB_USER', 'unittest')
+        pwd = os.environ.get('COUCHDB_PASSWORD', 'unittestpassword')
+        resp = requests.put(
+            'http://{0}/{1}'.format(admin_ep, user),
```

Comment:
Sorry @tomblench, I read your previous comment as a suggestion for just the Cloudant tests and not the CouchDB tests as well.  That has already been fixed and one can specify a URL for the Cloudant tests which will allow them to use either http or https.  The CouchDB tests have not changed based on that comment.  However, after seeing the comment from @ricellis I'll be adding something similar to the CouchDB tests.